src/service/resources-impl.js

```diff
@@ -1402,6 +1408,37 @@ export class Resources {
       // Selection API not supported.
     }
   }
+
+  /**
+   * Cleanup task queues from tasks for elements that has been unloaded.
+   * @param resource
+   * @private
+   */
+  maybeRemoveFromQueues_(resource) {
+    if (resource.getState() == ResourceState_.NOT_LAID_OUT) {
+      // If the layout promise for this resource has not resolved yet, remove
+      // it from the task queues to make sure this resource can be rescheduled
+      // for layout again later on.
+      // TODO(mkhatib): Look into if we need to do the same for preload tasks, this
+      // depends on whether we expect people to unload prerendered DOM.
+      // The answer might be no since we probably want to keep the prerendered
+      // DOM around to use as placeholder when the user navigate back to the
+      // document.
+      const taskId = resource.getTaskId(LAYOUT_TASK_ID_);
+      const scheduledTask = this.queue_.getTaskById(taskId);
+      const executingTask = this.exec_.getTaskById(taskId);
+      if (scheduledTask && this.queue_.dequeue(scheduledTask)) {
+        dev.fine(TAG_,
+            'layout promise ejected from queue after unload',
+            resource.debugid);
+      }
+      if (executingTask && this.exec_.dequeue(executingTask)) {
+        dev.fine(TAG_,
+            'layout promise ejected from exec queue after unload',
+            resource.debugid);
+      }
+    }
+  }
```

Comment:
I was looking for that queue. But it seems we're not using the `TaskQueue_` implementation for that one. It seems that it's just a normal JS array, is that correct?